Feat/io2021 c 64 mock data generation

Pinioo opened the pull request:
- Generating fake users
- Generating fake leaves
- List<Permission> in User changed to Permission

Pinioo committed:
Mock data generation
- Added javafaker library for mocking data
- Generating random Users, PersonalDatas and Leaves

Pinioo committed:
FakeUserLoader improvements
- refactored code
- user's leaves do not overlap now

Petroniuss reviewed (commented):
Quick question, it'd actually be helpful to have a known, fixed email per each user type. It'll probably make our lives easier. (I mean in addition to those fake ones)

Pinioo committed:
Fixed users with specific roles added
All passwords are: passw0rd
- Admin -> admin@gmail.com
- Supervisor -> supervisor@gmail.com
- User -> user@gmail.com

Pinioo commented:
> Quick question, it'd actually be helpful to have a known, fixed email per each user type. It'll probably make our lives easier. (I mean in addition to those fake ones)

Done. [admin@gmail.com; supervisor@gmail.com; user@gmail.com]
